tests/ipython/test_ipython.py

```diff
@@ -266,3 +247,56 @@ def test_line_magic_with_invalid_arguments(self, mocker, ipython):
             UsageError, match=r"unrecognized arguments: --invalid_arg=dummy"
         ):
             ipython.magic("reload_kedro --invalid_arg=dummy")
+
+
+class TestProjectPathResolution:
+    def test_only_path_specified(self):
+        result = _resolve_project_path(path="/test")
+        expected = Path("/test").resolve()
+        assert result == expected
+
+    def test_only_local_namespace_specified(self):
+        # pylint: disable=too-few-public-methods
+        class Context:
+            _project_path = Path("/test").resolve()
```

Comment:
The context for this change is that I moved to storing the project path in `context` (an instance of `KedroContext`), which is a variable in the IPython local namespace. This is the way that Viz does it and it is much safer than storing it in the 'top-level' of the IPython namespace, where a user can accidentally overwrite it.

This means that I had to change the testing implementation, I am using a dummy `Context` class rather than importing `KedroContext` and instantiating this, since its `__init__` takes a lot of arguments that we do not need here.  Do you think I should do the latter instead?

Reply:
Thanks for explaining! I'm happy to leave it like this, but perhaps add a small comment in the tests with this `Context` class explaining why it's there. I didn't immediately understand it was the `KedroContext` and not some completely different context used.

Reply:
Great, thanks for taking another look :). Made those changes and will merge